src/github.com/couchbase/sync_gateway/db/change_listener.go

```diff
@@ -186,3 +187,16 @@ func (waiter *changeWaiter) CurrentUserCount() uint64 {
 	}
 	return waiter.listener.CurrentCount(waiter.userKeys)
 }
+
+// Updates the set of channel keys in the ChangeWaiter (maintains the existing set of user keys)
+func (waiter *changeWaiter) UpdateChannels(chans channels.TimedSet) {
+	updatedKeys := make([]string, 0, 5)
```

Comment:
why is the capacity set to 5?  it seems like it should be set to `len(chans) + len(waiter.userKeys)` to minimize the amount of times the slice needs to be resized.